sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysisTool.java

```diff
@@ -176,22 +217,46 @@ public void run(CommandLine line, ToolRunningContext context) throws Exception {
         context.getOutputStream().println("Running analysis...");
         Stopwatch stopwatch = Stopwatch.createStarted();
         try (ComputationManager computationManager = DefaultComputationManagerConfig.load().createLongTimeExecutionComputationManager()) {
-            if (csv) {
-                try (Writer writer = Files.newBufferedWriter(outputFile, StandardCharsets.UTF_8);
-                     TableFormatter formatter = SensitivityValueCsvWriter.createTableFormatter(writer)) {
-                    SensitivityValueWriter valuesWriter = new SensitivityValueCsvWriter(formatter, contingencies);
-                    SensitivityAnalysis.run(network, network.getVariantManager().getWorkingVariantId(),
-                            factorsReader, valuesWriter, contingencies, variableSets, params,
-                            computationManager, Reporter.NO_OP);
+
+            if (line.hasOption(OUTPUT_UNIFIED)) {
+                if (csv) {
+                    throw new PowsyblException("Unsupported " + OUTPUT_UNIFIED + " option does not support csv file as argument of " + OUTPUT_FILE_OPTION + ". Must be json.");
                 }
+                List<SensitivityFactor> factors = new ArrayList<>();
+                factorsReader.read((functionType, functionId, variableType, variableId, variableSet, contingencyContext) -> {
+                    factors.add(new SensitivityFactor(functionType, functionId, variableType, variableId, variableSet, contingencyContext));
+                });
+                SensitivityAnalysisResult result = SensitivityAnalysis.run(network, network.getVariantManager().getWorkingVariantId(),
+                        factors, contingencies, variableSets, params,
+                        computationManager, Reporter.NO_OP);
+                ObjectMapper sensiObjectMapper = JsonUtil.createObjectMapper().registerModule(new SensitivityJsonModule());
+                JsonUtil.writeJson(outputFile, result, sensiObjectMapper);
```

Comment:
So when single output is chosen, we have a larger memory consumption. Why not, but as contingency status are very light data, it would have been possible to bufferered all and write at the end (which is not possible with values as we can have millions of them)

Reply:
Yes for efficiency writer based API will remain the best.